src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,74 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
+    {
+        private static readonly TypeWithAnnotations s_instance = TypeWithAnnotations.Create(new PlaceholderTypeArgumentSymbol());
+
+        public static ImmutableArray<TypeWithAnnotations> CreateTypeArguments(ImmutableArray<TypeParameterSymbol> typeParameters)
+        {
+            var result = ArrayBuilder<TypeWithAnnotations>.GetInstance();
+            foreach (var typeParameter in typeParameters)
+            {
+                result.Add(s_instance);
+            }
+            return result.ToImmutableAndFree();
+        }
+
+        private PlaceholderTypeArgumentSymbol(TupleExtraData? tupleData = null)
+            : base(tupleData)
+        {
+        }
+
+        protected override NamedTypeSymbol WithTupleDataCore(TupleExtraData newData)
+        {
+            return new PlaceholderTypeArgumentSymbol(newData);
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return string.Empty;
+            }
+        }
+
+        internal override bool MangleName
+        {
+            get
+            {
+                Debug.Assert(Arity == 0);
+                return false;
+            }
+        }
+
+        internal override DiagnosticInfo? ErrorInfo
+        {
+            get
+            {
+                return null;
+            }
+        }
+
+        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null)
+        {
+            if ((object)t2 == (object)this)
+            {
+                return true;
+            }
+
+            return t2 is PlaceholderTypeArgumentSymbol;
+        }
+
+        public override int GetHashCode()
+        {
+            return 0;
```

Comment:
>return 0; [](start = 12, length = 9)

Does `GetHashCode()` need to be overridden? It looks like the base implementation should be sufficient. 

If so, consider `return RuntimeHelpers.GetHashCode(this);` instead.